File: src/app/core/types/product.ts
export class Dimension {
  dimension: string;
  price: number;
}

export class Product {
  id: number;
  title: string;
  code: string;
  description: string;
  price: number;
  orderNumber: number;
  discount: number;
  discountDate: string;
  collectionId: number;
  categoryId: number;
  dimensions: Dimension[];
  width: number;
  height: number;
  depth: number;
  numberOnSite: number;
  drawingImage: string;
  drawingImageUrl: string;
  drawingImageExtension: string;
  images: any[];
  newImages: any[];
  updatedImages: any[];
  deletedImages: any[];
  count?: number;
  dimension?: string;
  bindLabel?: string;

  constructor(id: number, title: string, code: string, description: string, price: number, orderNumber: number,
    discount: number, discountDate: string, collectionId: number, categoryId: number, dimensions: Dimension[], width: number,
    height: number, numberOnSite: number, depth: number, images: any[], drawingImage: string, drawingImageUrl: string,
    drawingImageExtension: string) {
    this.id = id;
    this.title = title;
    this.code = code;
    this.price = price;
    this.orderNumber = orderNumber;
    this.description = description;
    this.discount = discount;
    this.discountDate = discountDate;
    this.collectionId = collectionId;
    this.dimensions = dimensions;
    this.width = width;
    this.height = height;
    this.numberOnSite = numberOnSite;
    this.depth = depth;
    this.images = images;
    this.drawingImage = drawingImage;
    this.drawingImageUrl = drawingImageUrl;
    this.drawingImageExtension = drawingImageExtension;
    this.categoryId = categoryId;
  }
}

File: backend/Marbomil.Infrastructure/Entities/Database/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Marbomil.Infrastructure.Entities.Database
{
    public class Product : IDbEntity
    {
        public int Id { get; set; }

        [MaxLength(100)]
        public string Title { get; set; }

        [MaxLength(50)]
        public string Code { get; set; }

        public int? OrderNumber { get; set; }
        public string Description { get; set; }

        [MaxLength(50)]
        public string Dimension { get; set; }

        public decimal Price { get; set; }
        public decimal Discount { get; set; }
        public DateTime? DiscountDate { get; set; }

        public int? CollectionId { get; set; }
        [ForeignKey("CollectionId")]
        public virtual Collection Collection { get; set; }

        public int CategoryId { get; set; }
        [ForeignKey("CategoryId")]
        public virtual Category Category { get; set; }

        public int NumberOnSite { get; set; }

        [MaxLength(1000)]
        public string DrawingImageUrl { get; set; }

        public virtual ICollection<ProductImage> Images { get; set; }
    }
}
